src/map/status.cpp

```diff
@@ -15256,11 +15260,15 @@ static int status_natural_heal(struct block_list* bl, va_list args)
 	if (flag&RGN_HP) {
 		// Interval to next recovery tick
 		rate = (int)(battle_config.natural_healhp_interval / (regen->rate.hp/100. * multi));
-		if (ud && ud->walktimer != INVALID_TIMER)
+		// Half recovery while moving only applies to players with certain traits
+		if (sd && ud && ud->walktimer != INVALID_TIMER)
```

Comment:
switch by bl type would be nice

Reply:
Is that really better? I feel a switch is only needed if you couldn't see everything in one screen.